plugins/crate-copy-azure/src/test/java/io/crate/copy/azure/AzureCopyIntegrationTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.copy.azure;
+
+import static io.crate.testing.Asserts.assertThat;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Locale;
+
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.IntegTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.github.dockerjava.api.model.ExposedPort;
+import com.github.dockerjava.api.model.HostConfig;
+import com.github.dockerjava.api.model.PortBinding;
+import com.github.dockerjava.api.model.Ports;
+
+public class AzureCopyIntegrationTest extends IntegTestCase {
+
+    private static final String CONTAINER_NAME = "test";
+    private static final String AZURITE_ACCOUNT = "devstoreaccount1";
+    private static final String AZURITE_KEY =
+            "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==";
+
+    private String endpoint;
+    private GenericContainer<?> azureContainer;
+    private BlobServiceClient blobServiceClient;
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        var plugins = new ArrayList<>(super.nodePlugins());
+        plugins.add(AzureCopyPlugin.class);
+        return plugins;
+    }
+
+    @Before
+    public void setup() {
+        azureContainer = new GenericContainer<>("mcr.microsoft.com/azure-storage/azurite:3.31.0")
```

Comment:
If we start using containers we'd need to update the dev docs to mention the docker/podman(?) dependency.

Is the existing azure stub server that we have not good enough?